### DISCUSSION TITLE
Visium coordinates different between SpatialFeaturePlot and seurat_object@image

### DISCUSSION THREAD

**Discussion by nrosewick** [satijalab/seurat#8570] [Q&A]
I plotted spatial spots from visium experiment using both SpatialFeaturePlot and using seurat @images coordinates. It seems that the y axis is inverted in brain@images . Is there some explanation ? 

Thank you

```
brain <- SeuratData::LoadData("stxBrain", type = "anterior1")

SpatialFeaturePlot(brain,"nCount_Spatial") |

brain@images$anterior1@coordinates %>%
  bind_cols(brain@meta.data) %>%
     filter(tissue==1) %>%
     mutate(x=imagecol,y=imagerow) %>%
     ggplot(aes(x=x,y=y)) +
     geom_point(aes(col=nCount_Spatial)) +
     scale_color_distiller(palette = "Spectral")
```

![image](https://github.com/satijalab/seurat/assets/3601374/e6481cb2-b483-460d-bd9e-8bfc11d655b4)

**Comment by bshihlab** [comment]
I found that by switching the imagerow and imagecol and flipping the y axis order, it would be back in the same orientation as the image

```
brain@images$anterior1@coordinates %>%
  bind_cols(brain@meta.data) %>%
     filter(tissue==1) %>%
     mutate(x=imagerow,y=imagecol) %>%
     ggplot(aes(x=x,y=y)) +
     geom_point(aes(col=nCount_Spatial)) +
     scale_color_distiller(palette = "Spectral") +
     scale_y_reverse() + 
     coord_fixed(ratio = 1, xlim = NULL, ylim = NULL, expand = TRUE, clip = "on") 
```

**Comment by rharao** [comment]
> Is there some explanation?

The usual convention for scatter charts is that the positive x direction is right from the origin and the positive y direction is **up** from the origin. ggplot2 is designed for statistical visualization and follows this convention.

The usual convention for raster images is that the origin is at the top left and pixel coordinates are non-negative, meaning that the positive x direction is right from the origin and the positive y direction is **down** from the origin. Visium spots are numbered analogously to pixels in a raster image.

As @bshihlab mentioned, if you want to make a ggplot2 visualization of the slide, you will have to add `scale_y_reverse()` to the plot to arrange the spot coordinates in the correct orientation. `SpatialFeaturePlot` takes care of this for you by calling `GeomSpatial`, which transforms the y coordinates as expected.
